src/runtime_src/core/common/query_requests.h

```diff
@@ -1184,6 +1185,50 @@ struct clock_freq_topology_raw : request
   get(const device*) const = 0;
 };
 
+struct xrt_resource_raw : request
+{
+  enum resource_type {
```

Comment:
No plain enums.  Use enum class instead.
xrt_resource_raw is a rather generic name, its fine, but if it is ipu specific, which the enum suggests, then please document this.  In general the structs in this file should be documented, e.g. what exactly is an xrt_resource_raw query?

Reply:
fixed.